lib/src/git.rs

```diff
@@ -449,6 +451,7 @@ pub fn import_refs(
 pub fn import_some_refs(
     mut_repo: &mut MutableRepo,
     git_settings: &GitSettings,
+    sync_settings: &SyncSettings,
```

Comment:
nit: I think the auto-track field can be copied to `GitSettings`. It's not wrong to load `GitSettings` from multiple sources.